pkg/nsx/services/common/services.go

```diff
@@ -30,8 +30,8 @@ type SubnetServiceProvider interface {
 	GetSubnetsByIndex(key, value string) []*model.VpcSubnet
 	CreateOrUpdateSubnet(obj client.Object, vpcInfo VPCResourceInfo, tags []model.Tag) (string, error)
 	GenerateSubnetNSTags(obj client.Object) []model.Tag
-	LockSubnet(path *string)
-	UnlockSubnet(path *string)
+	LockSubnet(path *string, read bool)
```

Comment:
Just a personal opinion, I prefer creating new methods, RWLockSubnet instead of passing in arguments, it is more straightforward.

Reply:
Updated, thanks!